src/lua/MapApi.cpp

```diff
@@ -1810,8 +1809,7 @@ int LuaContext::map_api_get_entities_in_rectangle(lua_State* l) {
     );
 
     lua_newtable(l);
-    for (auto it = entities.begin(); it != entities.end(); ++it) {
-      EntityPtr entity = *it;
+    for (auto&& entity: entities) {
```

Comment:
Use const MapEntityPtr& entity, not auto&&.
auto hides the actual type, so normally I only use it for verbose an uninformative types like iterator types.